### PR TITLE
Prevent crash in fixserialization when ClassSymbol.sourcefile is null.

### PR DESCRIPTION
**Pull request by lazaroclapp**
In general, neither `ClassSymbol.sourcefile` nor `ClassSymbol.classfile` are guaranteed to be non-null. This can lead to crashes when performing fixserialization, in particular when the symbol comes from a jar file.

This PR changes our logic to provide best effort file locations for code, checking first for `sourcefile`, then `classfile`, then defaulting to the string `"null"`. 

We also add a test case to simulate the behavior of other build systems (namely buck) where `sourcefile` might be `null`. This requires adding significant new mockito magic to our tests, including `mockito-inline`.

### CODE REVIEW

**Comment by lazaroclapp** [comment]
@nimakarimipour : A more robust version of the fix we did in our Zoom call, with tests (I actually decided against making artificial URIs once it became clear that this value is only really used in a couple places and the field can be nullable). Let me know if this looks like it fully addresses the issue.

@msridhar Mostly looping you in for the `mockito-inline` dependency. Not sure what's most important, having a test case covering this potential behavior of different build systems or keeping our test infrastructure relatively simple. Interestingly, this is the most elaborated mocking in our test suite so far.

**Review by msridhar — APPROVED** [review]
The mocking stuff is ok with me. Probably @nimakarimipour should look as well

File: nullaway/src/test/java/com/uber/nullaway/NullAwaySerializationTest.java [hunk 1]
```diff
@@ -1357,4 +1368,154 @@ public void errorSerializationTestLocalClassField() {
...
+                  "com.uber.nullaway.testdata.unannotated.MinimalUnannotatedClass",
+                  "null")) // <- ! the important bit
+          .setFactory(fixDisplayFactory)
+          .setOutputFileNameAndHeader(SUGGEST_FIX_FILE_NAME, SUGGEST_FIX_FILE_HEADER)
+          .doTest();
+    }
+  }
+
+  @Test
+  public void suggestNullableArgumentOnBytecodeClassFileInfoOnly() {
```
Comment (nimakarimipour) [inline nullaway/src/test/java/com/uber/nullaway/NullAwaySerializationTest.java:1466]: Nice test 👍🏻

**Review by nimakarimipour — APPROVED** [review]
Finished review, this looks great to me. 🙂
